main/mafia.hs

```diff
@@ -502,6 +509,17 @@ mafiaScript :: File -> [Argument] -> EitherT MafiaError IO ()
 mafiaScript file args =
   firstT MafiaScriptError $ runScript file args
 
+mafiaExec :: [Argument] -> EitherT MafiaError IO ()
+mafiaExec args =
+  liftCabal $ cabal_ "exec" $ fixupArgs args
+  where
+    fixupArgs = \case
+      [] -> [] -- Should not happen.
+      [x] -> [x] -- A command without arguments.
+      -- Insert `--` between the command and its arguments so cabal doesn't
+      -- mess with them.
+      (x:xs) -> x : "--" : xs
```

Comment:
To run `sh -c 'echo Hello'` with cabal you would do:
```
cabal exec sh -- -c 'echo Hello'
```
so that cabal knows not to process command line args after `--`. If I don't insert `--`, cabal will treat the `-c` as one of the cabal arguments.